src/state/opamSysInteract.ml

```diff
@@ -253,31 +253,26 @@ let packages_status packages =
           | Some (pkg, _repo) -> pkg
           | None -> s)
       |> (fun l -> run_query_command "apk" ("policy"::l))
-      (* fst_version is here to do not add package while their informations
-         parsing it not yet finished, or at least a minimum complete *)
-      |> List.fold_left (fun (current, fst_version, installed, repo, instavail) l ->
-          try
+      |> List.fold_left (fun (pkg, installed, instavail) l ->
+          try (* package name *)
             let new_pkg = Re.(Group.get (exec pkg_name l) 1) in
-            let pkg = match repo with Some r -> current^"@"^r | None -> current in
-            let instavail = add_pkg fst_version pkg installed instavail in
-            new_pkg, true, false, None, instavail
+            new_pkg, false, instavail
           with Not_found ->
             if l.[2] != ' ' then (* only version in after two spaces *)
-              let pkg = match repo with Some r -> current^"@"^r | None -> current in
-              let instavail = add_pkg fst_version pkg installed instavail in
-              current, false, false, None, instavail
-            else
-            if Re.execp is_installed l then
-              current, fst_version, true, repo, instavail
+              pkg, false, instavail
+            else if Re.execp is_installed l then
```

Comment:
I can't add a suggestion on unchanged code above but the definition for `is_installed` could be a bit more precise.
Currently, according to the code source:
- https://git.alpinelinux.org/apk-tools/tree/src/app_policy.c
- https://git.alpinelinux.org/apk-tools/tree/src/database.c#n58
Only `lib/apk/db/installed` is possible there to let the user know the package is installed, and anything else (including something like `https://dl-cdn.alpinelinux.org/alpine/v3.13/installed`) is possible to list a repository.